src/reformatters/common/dynamical_dataset.py

```diff
@@ -25,6 +28,11 @@ class DynamicalDataset(FrozenBaseModel, Generic[DATA_VAR, SOURCE_FILE_COORD]):
     template_config: TemplateConfig[DATA_VAR]
     region_job_class: type[RegionJob[DATA_VAR, SOURCE_FILE_COORD]]
 
+    @computed_field  # type: ignore[prop-decorator]
+    @property
+    def id(self) -> str:
+        return self.template_config.dataset_id
```

Comment:
I didn't love having to do `app.add_typer(dataset.create_cli(), name=dataset.template_config.dataset_id)` but it wouldn't be the end of the world if we want to avoid continually promoting this value up the tree.